eogrow/pipelines/zipmap.py

```diff
@@ -0,0 +1,133 @@
+                params_model: BaseSchema = import_object(values["params_model"])
+                return params_model.parse_obj(v).dict()
+            return v
+
+        output_folder_key: str = Field(
+            description="The storage manager key pointing to the output folder for the algorithm pipeline."
+        )
+        output_feature: FeatureSpec
+
+        compress_level: int = Field(1, description="Level of compression used in saving eopatches.")
+
+    config: Schema
+
+    def filter_patch_list(self, patch_list: List[str]) -> List[str]:
+        """EOPatches are filtered according to existence of new features"""
+        # Note: does not catch missing BBox or Timestamp
+        filtered_patch_list = get_patches_with_missing_features(
+            self.storage.filesystem,
+            self.storage.get_folder(self.config.output_folder_key),
+            patch_list,
+            [self.config.output_feature],
+        )
+
+        return filtered_patch_list
+
+    def get_load_nodes(self) -> List[EONode]:
+        """Prepare all nodes with load tasks."""
+        load_schema: DefaultDict[str, Set[FeatureSpec]] = defaultdict(set)
+        for input_feature in self.config.input_features:
+            features_to_load = load_schema[input_feature.folder_key]
+            features_to_load.add(input_feature.feature)
+
+            if input_feature.include_bbox_and_timestamp:
+                features_to_load.add(FeatureType.BBOX)
+                if input_feature.feature[0].is_temporal():
+                    features_to_load.add(FeatureType.TIMESTAMP)
+
+        load_nodes = []
+        for folder_key, features in load_schema.items():
+            folder_path = self.storage.get_folder(folder_key, full_path=True)
+            load_nodes.append(
+                EONode(LoadTask(folder_path, config=self.sh_config, features=list(features), lazy_loading=True))
+            )
+
+        return load_nodes
+
+    def get_zipmap_node(self, previous_node: EONode) -> EONode:
+        """Adds all algorithm and dataframe-saving nodes and returns the endnode."""
+
+        zipmap = import_object(self.config.zipmap_import_path)
+        input_features = [input_schema.feature for input_schema in self.config.input_features]
+        zipmap_task = ZipFeatureTask(input_features, self.config.output_feature, zipmap, **self.config.params)
+
+        return EONode(zipmap_task, inputs=[previous_node])
+
+    def build_workflow(self) -> EOWorkflow:
+        """Builds the workflow"""
+
+        load_nodes = self.get_load_nodes()
+        merge_node = EONode(MergeEOPatchesTask(), inputs=load_nodes)
+
+        mapping_node = self.get_zipmap_node(merge_node)
+
+        save_path = self.storage.get_folder(self.config.output_folder_key, full_path=True)
+        save_task = SaveTask(
+            save_path,
+            config=self.sh_config,
+            features=[self.config.output_feature, FeatureType.BBOX, FeatureType.TIMESTAMP],
```

Comment:
Will this work if the bbox/timestamp are not loaded on input?

Reply:
yes, I also had to test this cause I didn't remember
for saving this parameter serves as a filter, and not precise specification